pandas/core/categorical.py

```diff
@@ -443,12 +443,18 @@ def _validate_categories(cls, categories):
             raise ValueError('Categorical categories must be unique')
         return categories
 
-    def _set_categories(self, categories):
+    def _set_categories(self, categories, validate=True):
         """ Sets new categories """
```

Comment:
`fastpath` here goes thru `set_categories` which calls `._set_categories`. It still needs to go through the `validate` part so I'm not really sure what it's fastpathing.

All the warnings are caught in the `test_categorical`. There were quite a few.